craft_parts/state_manager/reports.py

```diff
@@ -134,7 +134,9 @@ def reason(self) -> str:
             if reasons_count > 1 or len(self.changed_dependencies) > 1:
                 reasons.append("dependencies")
             else:
-                reasons.append(f"{self.changed_dependencies[0].part_name!r}")
+                part_name = self.changed_dependencies[0].part_name
+                step_name = self.changed_dependencies[0].step.name.lower()
+                reasons.append(f"part {part_name!r} {step_name}")
```

Comment:
Is there anything else aside from "changed"?

```suggestion
                reasons.append(f"{step_name} for part {part_name!r}")
```

Reply:
It's only that for now, we can improve it in future iterations.